cdap-ui/app/features/workflows/controllers/tabs/schedule-ctrl.js

```diff
@@ -1,87 +1,92 @@
+        angular.forEach(this.schedules, schedule => {
+          if (schedule.scheduleType === 'TIME') {
+            var parse = schedule.schedule.cronExpression.split(' ');
+            schedule.time = {};
+            schedule.time.min = parse[0];
+            schedule.time.hour = parse[1];
+            schedule.time.day = parse[2];
+            schedule.time.month = parse[3];
+            schedule.time.week = parse[4];
 
-            myWorkFlowApi.schedulesPreviousRunTime(params)
+            this.myWorkFlowApi.schedulesPreviousRunTime(params)
               .$promise
-              .then(function(timeResult) {
+              .then( timeResult => {
                 if (timeResult[0]) {
-                  v.lastrun = timeResult[0].time;
+                  schedule.lastrun = timeResult[0].time;
                 } else {
-                  v.lastrun = 'NA';
+                  schedule.lastrun = 'NA';
                 }
               });
           } else {
-            v.lastrun = 'NA';
+            schedule.lastrun = 'NA';
           }
-          v.isOpen = false;
-          myWorkFlowApi.pollScheduleStatus({
-            appId: $state.params.appId,
-            scheduleId: v.schedule.name,
-            scope: $scope
+          schedule.isOpen = false;
+          this.myWorkFlowApi.pollScheduleStatus({
+            appId: this.$state.params.appId,
+            scheduleId: schedule.schedule.name,
+            scope: this.$scope
           })
             .$promise
-            .then(function(response) {
-              v.status = response.status;
+            .then( response => {
+              schedule.status = response.status;
             });
         });
 
         if (this.schedules.length > 0) {
           this.schedules[0].isOpen = true;
         }
-      }.bind(this));
-
-    this.suspendSchedule = function (obj) {
-      obj.status = 'SUSPENDING';
-
-      myWorkFlowApi.scheduleSuspend({
-        appId: $state.params.appId,
-        scheduleId: obj.schedule.name,
-        scope: $scope
-      }, {},
-      function success() {},
-      function error(err) {
-        myAlert({
-          title: 'Cannot Suspend Schedule',
-          content: err
-        });
       });
-    };
+  }
 
-    this.resumeSchedule = function (obj) {
-      obj.status = 'RESUMING';
-
-      myWorkFlowApi.scheduleResume({
-        appId: $state.params.appId,
-        scheduleId: obj.schedule.name,
-        scope: $scope
-      }, {},
-      function success() {},
-      function error(err) {
-        myAlert({
-          title: 'Cannot Resume Schedule',
-          content: err
-        });
+  suspendSchedule(obj) {
+    this.myWorkFlowApi.scheduleSuspend({
+      appId: this.$state.params.appId,
+      scheduleId: obj.schedule.name,
+      scope: this.$scope
+    }, {},
+    function success() {},
+    function error(err) {
+      this.myAlert({
+        title: 'Cannot Suspend Schedule',
+        content: err
       });
-    };
+    }.bind(this));
```

Comment:
Only fat arrow functions ```()=>``` don't need the executing scope. Apart from that if we are using ```function () {..}``` syntax we need to bind it with a scope if it needs to be executed in a specific scope.